Cannot Generate Cypress Project on 13.10

## Current Behavior
Error generating cypress project. Getting error related to not being able to find the logger.


## Expected Behavior
Be able to geenrate a cypress e2e project. This is the first time I've run this command.

## Steps to Reproduce
Install `@nrwl/cypress`

`yarn add --dev @nrwl/cypress`
`yarn nx generate @nrwl/cypress:cypress-project <project>-e2e --project=<project>`

package.json:

```json
{
  "name": "<name>",
  "version": "0.0.0",
  "private": true,
  "workspaces": [
    "apps/*",
    "packages/*",
    "tools/*"
  ],
  "devDependencies": {
    "@babel/preset-env": "^7.16.11",
    "@babel/preset-react": "^7.16.7",
    "@codebrew/nx-aws-cdk": "^1.0.3",
    "@craco/craco": "7.0.0-alpha.3",
    "@nrwl/cli": "^13.10.0",
    "@nrwl/cypress": "^13.10.0",
    "@nrwl/eslint-plugin-nx": "13.9.7",
    "@nrwl/jest": "13.9.7",
    "@nrwl/linter": "13.9.7",
    "@nrwl/react": "^13.9.7",
    "@nrwl/web": "13.9.7",
    "@nrwl/workspace": "13.10.0",
    "@testing-library/dom": "^8.13.0",
    "@testing-library/jest-dom": "^5.16.4",
    "@testing-library/react": "12.1.4",
    "@testing-library/react-hooks": "7.0.2",
    "@testing-library/user-event": "^13.5.0",
    "@tsconfig/node14": "^1.0.1",
    "@types/chance": "^1.1.3",
    "@types/graphql": "^14.5.0",
    "@types/jest": "27.0.2",
    "@types/node": "^17.0.23",
    "@types/react": "17.0.40",
    "@types/react-dom": "17.0.13",
    "@typescript-eslint/eslint-plugin": "~5.10.0",
    "@typescript-eslint/parser": "~5.10.0",
    "autoprefixer": "^10.4.4",
    "aws-sdk": "^2.1095.0",
    "babel-jest": "27.2.3",
    "babel-loader": "^8.2.4",
    "babel-webpack-plugin": "^0.1.2",
    "chalk": "^5.0.0",
    "chance": "^1.1.8",
    "craco-babel-loader": "^1.0.3",
    "cypress": "^9.1.0",
    "dotenv": "^16.0.0",
    "eslint": "^8.12.0",
    "eslint-config-prettier": "8.1.0",
    "eslint-config-react-app": "^7.0.0",
    "eslint-plugin-": "nrwl/nx",
    "eslint-plugin-cdk": "^1.3.0",
    "eslint-plugin-cypress": "^2.10.3",
    "eslint-plugin-import": "2.25.4",
    "eslint-plugin-jsx-a11y": "6.5.1",
    "eslint-plugin-react": "7.29.3",
    "eslint-plugin-react-hooks": "4.3.0",
    "install-peerdeps": "^3.0.3",
    "jest": "27.2.3",
    "jest-console": "^0.1.0",
    "jest-junit": "^13.0.0",
    "jest-watch-typeahead": "^1.0.0",
    "mocha": "5.2.0",
    "mochawesome-report-generator": "^6.2.0",
    "nx": "^13.10.0",
    "postcss": "^8.4.12",
    "prettier": "^2.5.1",
    "react-scripts": "5",
    "react-test-renderer": "17.0.2",
    "ts-jest": "27.0.5",
    "typescript": "^4.6.3",
    "web-vitals": "^2.1.4",
    "webpack": "^5.72.0"
  },
  "engines": {
    "npm": ">=7.0.0",
    "node": ">=14.0.0"
  },
  "packageManager": "npm@8.5.0"
}
```

### Failure Logs
```sh
Cannot find module 'nx/src/shared/logger'
Require stack:
- /Users/victor/dev/workspace/coquimono/node_modules/@nrwl/cypress/node_modules/@nrwl/devkit/index.js
- /Users/victor/dev/workspace/coquimono/node_modules/@nrwl/cypress/src/generators/cypress-project/cypress-project.js
- /Users/victor/dev/workspace/coquimono/node_modules/nx/src/config/workspaces.js
- /Users/victor/dev/workspace/coquimono/node_modules/nx/src/command-line/generate.js
- /Users/victor/dev/workspace/coquimono/node_modules/nx/src/command-line/nx-commands.js
- /Users/victor/dev/workspace/coquimono/node_modules/nx/bin/init-local.js
- /Users/victor/dev/workspace/coquimono/node_modules/nx/bin/nx.js
```

### Environment
```
yarn run v1.22.18
$ /Users/victor/dev/workspace/coquimono/node_modules/.bin/nx report

 >  NX   Report complete - copy this into the issue template

   Node : 16.14.2
   OS   : darwin arm64
   yarn : 1.22.18
   
   nx : 13.10.0
   @nrwl/angular : Not Found
   @nrwl/cypress : 13.10.0
   @nrwl/detox : Not Found
   @nrwl/devkit : 13.10.0
   @nrwl/eslint-plugin-nx : 13.9.7
   @nrwl/express : Not Found
   @nrwl/jest : 13.9.7
   @nrwl/js : 13.10.0
   @nrwl/linter : 13.9.7
   @nrwl/nest : Not Found
   @nrwl/next : Not Found
   @nrwl/node : Not Found
   @nrwl/nx-cloud : Not Found
   @nrwl/nx-plugin : Not Found
   @nrwl/react : 13.10.0
   @nrwl/react-native : Not Found
   @nrwl/schematics : Not Found
   @nrwl/storybook : 13.10.0
   @nrwl/web : 13.9.7
   @nrwl/workspace : 13.10.0
   typescript : 4.6.3
   rxjs : 6.6.7
   ---------------------------------------
   Community plugins:
   	 @codebrew/nx-aws-cdk: 1.0.3

Done in 0.64s.
```

```
yarn run v1.22.18
$ /Users/<user>/dev/workspace/<repo>/node_modules/.bin/nx g @nrwl/nrwl:cypress-project --name=<project>-e2e --project=<project> --verbose
Error: Unable to resolve @nrwl/nrwl:cypress-project.
Cannot find module '@nrwl/nrwl/package.json'
Require stack:
- /Users/<user>/dev/workspace/<repo>/node_modules/nx/src/utils/nx-plugin.js
- /Users/<user>/dev/workspace/<repo>/node_modules/nx/src/config/workspaces.js
- /Users/<user>/dev/workspace/<repo>/node_modules/nx/src/command-line/generate.js
- /Users/<user>/dev/workspace/<repo>/node_modules/nx/src/command-line/nx-commands.js
- /Users/<user>/dev/workspace/<repo>/node_modules/nx/bin/init-local.js
- /Users/<user>/dev/workspace/<repo>/node_modules/nx/bin/nx.js
    at Workspaces.readGenerator (/Users/victor/dev/workspace/<repo>/node_modules/nx/src/config/workspaces.js:116:19)
    at Object.<anonymous> (/Users/victor/dev/workspace/<repo>/node_modules/nx/src/command-line/generate.js:119:83)
    at Generator.next (<anonymous>)
    at /Users/<user>/dev/workspace//<repo>/node_modules/tslib/tslib.js:117:75
    at new Promise (<anonymous>)
    at Object.__awaiter (/Users/victor/dev/workspace/<repo>/node_modules/tslib/tslib.js:113:16)
    at /Users/<user>/dev/workspace/<repo>/node_modules/nx/src/command-line/generate.js:116:68
    at /Users/<user>/dev/workspace/<repo>/node_modules/nx/src/utils/params.js:9:26
    at Generator.next (<anonymous>)
    at /Users/<user>/dev/workspace/<repo>/node_modules/tslib/tslib.js:117:75
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

--- a/packages/cypress/src/generators/cypress-project/cypress-project.spec.ts
+++ b/packages/cypress/src/generators/cypress-project/cypress-project.spec.ts
@@ -413,6 +413,23 @@ describe('schematic:cypress-project', () => {
           expect(projectConfig.tags).toEqual([]);
         });
       });
+
+      it('should not throw an error when --project does not have targets', async () => {
+        const projectConf = readProjectConfiguration(tree, 'my-app');
+        delete projectConf.targets;
+
+        updateProjectConfiguration(tree, 'my-app', projectConf);
+        await cypressProjectGenerator(tree, {
+          name: 'my-app-e2e',
+          project: 'my-app',
+          linter: Linter.EsLint,
+        });
+
+        const projectConfig = readProjectConfiguration(tree, 'my-app-e2e');
+        expect(projectConfig.targets['e2e'].options.devServerTarget).toEqual(
+          'my-app:serve'
+        );
+      });
     });
 
     describe('--linter', () => {

--- a/packages/cypress/src/generators/cypress-project/cypress-project.ts
+++ b/packages/cypress/src/generators/cypress-project/cypress-project.ts
@@ -13,6 +13,8 @@ import {
   Tree,
   updateJson,
   ProjectConfiguration,
+  stripIndents,
+  logger,
 } from '@nrwl/devkit';
 import { Linter, lintProjectGenerator } from '@nrwl/linter';
 import { runTasksInSerial } from '@nrwl/workspace/src/utilities/run-tasks-in-serial';
@@ -82,8 +84,15 @@ function addProject(tree: Tree, options: CypressProjectSchema) {
     };
   } else if (options.project) {
     const project = readProjectConfiguration(tree, options.project);
+
+    if (!project.targets) {
+      logger.warn(stripIndents`
+      NOTE: Project, "${options.project}", does not have any targets defined and a baseUrl was not provided. Nx will use  
+      "${options.project}:serve" as the devServerTarget. But you may need to define this target within the project, "${options.project}".
+      `);
+    }
     const devServerTarget =
-      project.targets.serve && project.targets.serve.defaultConfiguration
+      project.targets?.serve && project.targets?.serve?.defaultConfiguration
         ? `${options.project}:serve:${project.targets.serve.defaultConfiguration}`
         : `${options.project}:serve`;
     e2eProjectConfig = {

